src/tasks/taskPseudoTerminal.ts

```diff
@@ -0,0 +1,143 @@
+	}
+}
+
+async function spawnCommand (command: string, options: cp.SpawnOptions, onStdout: StdListeners, onStderr: StdListeners, token?: vscode.CancellationToken): Promise<void> {
+	return new Promise((resolve, reject) => {
+		let cancellationListener: vscode.Disposable;
+		options = options || {};
+		options.shell = true;
+
+		const process = cp.spawn(command, [], options);
+
+		process.stdout.on('data', (data: Buffer) => {
+			onStdout(data.toString());
+		});
+
+		process.stderr.on('data', (data: Buffer) => {
+			onStderr(data.toString());
+		});
+
+		process.on('close', (code, signal) => {
+			if (cancellationListener) {
+				cancellationListener.dispose();
+			}
+
+			if (token && token.isCancellationRequested) {
+				return reject(new Error('user cancelled'));
+			} else if (code) {
+				// throw nice error
+				const error = new CommandError(`Process exited with ${code}`, command, code, signal);
+
+				return reject(error);
+			}
+			return resolve();
+		});
+
+		if (token) {
+			cancellationListener = token.onCancellationRequested(() => {
+				process.kill();
+			});
+		}
+	});
+}
+
+export class TaskPseudoTerminal implements vscode.Pseudoterminal {
+	private readonly closeEmitter: vscode.EventEmitter<number> = new vscode.EventEmitter<number>();
+	private readonly cts: vscode.CancellationTokenSource = new vscode.CancellationTokenSource();
+	private readonly task: TitaniumTaskBase;
+	private readonly taskProvider: CommandTaskProvider;
+	private readonly writeEmitter: vscode.EventEmitter<string> = new vscode.EventEmitter<string>();
+
+	public readonly onDidWrite: vscode.Event<string> = this.writeEmitter.event;
+	public readonly onDidClose: vscode.Event<number> = this.closeEmitter.event;
+
+	public constructor (taskProvider: CommandTaskProvider, task: TitaniumTaskBase) {
+		this.taskProvider = taskProvider;
+		this.task = task;
+	}
+
+	public open (): void {
+		const folder = this.task.scope === vscode.TaskScope.Workspace
+			? vscode.workspace.workspaceFolders![0]
+			: this.task.scope as vscode.WorkspaceFolder;
+
+		const executionContext: TaskExecutionContext = {
+			cancellationToken: this.cts.token,
+			folder,
+			terminal: this,
+		};
+
+		ExtensionContainer.context.globalState.update(GlobalState.Running, true);
+		vscode.commands.executeCommand('setContext', GlobalState.Running, true);
+
+		this.taskProvider.executeTask(executionContext, this.task).then((result) => this.close(result));
+	}
+
+	public close (code?: number): void {
+		this.cts.cancel();
+		this.closeEmitter.fire(code || 0);
+
+		ExtensionContainer.context.globalState.update(GlobalState.Running, false);
+		vscode.commands.executeCommand('setContext', GlobalState.Running, false);
+	}
+
+	public async executeCommand (command: string, folder: vscode.WorkspaceFolder, token?: vscode.CancellationToken): Promise<void> {
+
+		this.write(`${command} \r\n\r\n`);
+
+		await spawnCommand(
+			command,
+			{ cwd: folder.uri.fsPath, env: { ...process.env, FORCE_COLOR: '1' } },
```

Comment:
<!--
  0 failure: 
  1 warning:  src/tasks/taskPse...
  
  
  DangerID: danger-id-Danger;
  File: src/tasks/taskPseudoTerminal.ts;
  Line: 116;
-->

- :warning: src/tasks/taskPseudoTerminal.ts line 116 – Rest/spread properties are not supported until Node.js 8.3.0. The configured version range is '>=8.0.0'. (node/no-unsupported-features/es-syntax)